cmd/cloud-controller-manager/main.go

```diff
@@ -31,12 +31,14 @@ import (
 	cloudprovider "k8s.io/cloud-provider"
 	"k8s.io/cloud-provider/app"
 	cloudcontrollerconfig "k8s.io/cloud-provider/app/config"
+	"k8s.io/cloud-provider/app/names"
 	"k8s.io/cloud-provider/options"
 	"k8s.io/component-base/cli"
 	cliflag "k8s.io/component-base/cli/flag"
 	_ "k8s.io/component-base/metrics/prometheus/clientgo" // load all the prometheus client-go plugins
 	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
 	"k8s.io/klog/v2"
+	kcmnames "k8s.io/kubernetes/cmd/kube-controller-manager/app/names"
```

Comment:
I'm pretty sure we don't want references to `k8s.io/kubernetes/cmd/kube-controller-manager` from cloud-controller-manager... should the nodeipam name be in k8s.io/cloud-provider/app/names instead?

Reply:
I would like to have the controller name  defined in the same place/module where the controller lives. And the nodeipam controller lives in KCM and is imported by CCM

https://github.com/kubernetes/kubernetes/blob/894cfdfeb4a09083849f9193f7c36360e61509eb/cmd/cloud-controller-manager/nodeipamcontroller.go#L37-L39

KCM imports CCM controllers/options/names as well. We prevent circular dependency by choosing the relevant packages for nodeipam controller and only importing them in the `main` package of CCM

https://github.com/kubernetes/kubernetes/blob/894cfdfeb4a09083849f9193f7c36360e61509eb/cmd/cloud-controller-manager/.import-restrictions#L4

Reply:
I'm not really sure if anyone uses CCM from K/K. It seems that it was some step in Cloud Provider Extraction/Migration that is not-used anymore.

@cheftako and @nckturner, do you know the exact context about this CCM implementation and if anyone uses it?

Reply:
Do we want to deprecate or remove the support for the binary and the relevant code in parallel to this PR?

Reply:
Yes, we can propose it in a PR.

Reply:
I am proposing a deprecation in https://github.com/kubernetes/kubernetes/pull/117564 - more details in the PR.

Reply:
So are we OK with this because it is going away?

Reply:
Some extra content: cmd/cloud-controller-manager previously stays as an example for new incoming cloud providers. Do we still plan to have a way to educate the incoming cloud provider if this is removed?

Reply:
Please correct me if I am wrong, but I do not see an issue with the imports as I mentioned in the earlier comment, even if the example CCM does not go away. 

We can start with the deprecation and have some time to discuss where to move it before the removal. We can also keep the README in cmd/cloud-controller-manager and reference the new location after that.